api/api.go

```diff
@@ -139,14 +154,14 @@ func (s *authedRequester) Do(r *Request) *Response {
 	if err != nil {
 		return &Response{err: err}
 	}
-	if res.StatusCode != 200 {
-		return &Response{err: fmt.Errorf(r.Endpoint + fmt.Sprintf(" Not ok: HTTP %d", res.StatusCode))}
-	}
 	defer res.Body.Close()
 	data, err := ioutil.ReadAll(res.Body)
 	if err != nil {
 		return &Response{err: err}
 	}
+	if res.StatusCode != 200 {
+		return &Response{err: fmt.Errorf(r.Endpoint + fmt.Sprintf(" Not ok: HTTP %d\n", res.StatusCode) + string(data[:]))}
```

Comment:
This wasn't your fault, but the `fmt.Sprintf` here is redundant inside a `fmt.Errorf`---`Errorf` is basically `Sprintf` into an `error`.  This should be

```go
return &Response{err: fmt.Errorf("%s Not ok: HTTP %d\n%s", r.Endpoint, res.StatusCode, string(data)) }
```

The `string()` on `data` might be redundant in that case, but I'm not sure, and my gut tells me it isn't.
```